api/app_boxes_fixtures_test.go

```diff
@@ -0,0 +1,622 @@
+package api
+
+import (
+	"encoding/base32"
+	"encoding/base64"
+	"fmt"
+	"testing"
+
+	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/data/transactions"
+	"github.com/algorand/go-algorand/data/transactions/logic"
+	"github.com/algorand/go-algorand/ledger"
+	"github.com/algorand/go-algorand/rpcs"
+	"github.com/algorand/indexer/processor"
+	"github.com/algorand/indexer/util/test"
+	"github.com/stretchr/testify/require"
+)
+
+func goalEncode(t *testing.T, s string) string {
+	b1, err := logic.NewAppCallBytes(s)
+	require.NoError(t, err, s)
+	b2, err := b1.Raw()
+	require.NoError(t, err)
+	return string(b2)
+}
+
+var goalEncodingExamples map[string]string = map[string]string{
+	"str":         "str",
+	"string":      "string",
+	"int":         "42",
+	"integer":     "100",
+	"addr":        basics.AppIndex(3).Address().String(),
+	"address":     basics.AppIndex(5).Address().String(),
+	"b32":         base32.StdEncoding.EncodeToString([]byte("b32")),
+	"base32":      base32.StdEncoding.EncodeToString([]byte("base32")),
+	"byte base32": base32.StdEncoding.EncodeToString([]byte("byte base32")),
+	"b64":         base64.StdEncoding.EncodeToString([]byte("b64")),
+	"base64":      base64.StdEncoding.EncodeToString([]byte("base64")),
+	"byte base64": base64.StdEncoding.EncodeToString([]byte("byte base64")),
+	"abi":         `(uint64,string,bool[]):[399,"pls pass",[true,false]]`,
+}
+
+func setupLiveBoxes(t *testing.T, proc processor.Processor, l *ledger.Ledger) {
```

Comment:
Admittedly, there are D.R.Y. violations in this test (and in the other boxes tests as well). However, in my opinion, violating D.R.Y. for testing isn't always a bad thing. It's easier for tests to evolve separately when they don't need to share common code, and the redundant code doesn't bloat the production binary given go's packaging  method. Also, getting past go's packaging and privacy policy can be daunting.